lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -518,6 +520,30 @@ data StakeDistribution = StakeDistribution
     , unassigned :: Quantity "lovelace" Word64
     } deriving (Eq, Show, Generic)
 
+-- | Information about a stake pool. This information is not used directly by
+-- cardano-wallet. It is sourced from the stake pool registry and passed
+-- straight through to API consumers.
+data StakePoolMetadata = StakePoolMetadata
+    { ticker :: StakePoolTicker
+    -- ^ Short human-readable ID for the stake pool.
+    , homepage :: Text
+    -- ^ Absolute URL for the stake pool's homepage link.
```

Comment:
**[suggestion: type-safety]**
Perhaps use [Network.URI](https://hackage.haskell.org/package/network-uri-2.6.1.0/docs/Network-URI.html) ?

**Why**
URIs have a quite well-defined format and we can rely on `parseURI` to parse the JSON string easily. It also makes it easier to then cross it with additional checks on the `uriScheme`.

Reply:
I originally did use parseURI then removed it.
It doesn't add anything, unless we must sanitise URLs that users might click on.

Reply:
I think it's fine for _now_ as we'll only connect to the CF repository. Later, we may want to allow different sources, potentially "bad", to be chosen by users. 

Let's keep it simple for now.